fs/sync/sync_test.go

```diff
@@ -59,6 +59,18 @@ func TestCopy(t *testing.T) {
 
 	fstest.CheckItems(t, r.Flocal, file1)
 	fstest.CheckItems(t, r.Fremote, file1)
+
+	// Test copy with source file that's updating
```

Comment:
The trouble with doing the test here is that this will be run by the integration tests with remotes which aren't local which will definitely fail...

Also that time.Sleep makes it racy so that sometimes it will fail.

I think the best place for this test might [be here](https://github.com/ncw/rclone/blob/master/backend/local/local_internal_test.go)

I'd test it by constructing a fake `localOpenFile` open on a temporary file you make, then do a `Read` - check OK, change the file, Check `Read` gives the error.  You'll need to make an local.Object with a few things in too (just make it by hand).

What do you think?

Reply:
That would be a better way to test. I tried to be clever with the goroutine but it fails 1/10, race condition. Updating tests in a few!